src/classes/PermissionLevels.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * A helper class for building valid permission Structures
+ */
+class PermissionLevels {
+
+  /**
+   * @typedef {object} permLevel
+   * @memberof PermissionLevels
+   * @property {boolean} break Whether the level should break (stop processing higher levels, and inhibit a no permission error)
+   * @property {Function} check The permission checking function
+   */
+
+  /**
+   * Creates a new PermissionLevels instance.
+   * Note that constructing any PermissionLevel instance will always insert a "Owner only" level for you
+   * at the highest number you specify automatically. You can still however remove this and change it after
+   * constructing.
+   * @param  {number} [size=10] The number of levels you want to allocate for this PermissionLevels instance
+   */
+  constructor(size = 10) {
+    const s = parseInt(size);
+    if (typeof s !== "number") throw new Error("Size must be a valid integer");
+
+    /**
+     * The number of permission levels allowed in this instance.
+     * Technically this will be the size you input + 1 since Array[0] would be Level 0
+     * @type {number}
+     */
+    this.size = s;
+
+    /**
+     * Cached array of levels that get used for determining permissions.
+     * @type {Array}
+     */
+    this.levels = new Array(s).fill({ break: false, check: () => false }, 0, s);
```

Comment:
Why are you using new Array()?

Reply:
So that I can construct the cached array with a finite amount of levels. You can't do that with just doing
```javascript
this.levels = [];
```

Reply:
Makes sense